weld/ast.rs

```diff
@@ -523,21 +522,22 @@ impl<T: TypeBounds> Expr<T> {
 
     /// Recursively transforms an expression in place by running a function on it and optionally replacing it with another expression.
     pub fn transform_and_continue<F>(&mut self, func: &mut F)
-        where F: FnMut(&mut Expr<T>) -> Option<(Expr<T>, bool)>
+        where F: FnMut(&mut Expr<T>) -> (Option<Expr<T>>, bool)
```

Comment:
Yup, the `bool` states whether to recurse down and the expression is what to replace it with. Before, erroring out was hard since there was no way of saying "don't replace the expression and don't recurse down."